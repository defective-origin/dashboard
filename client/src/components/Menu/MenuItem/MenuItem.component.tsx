import React from 'react'
import MuiMenuItem, {
  MenuItemProps as MuiMenuItemProps,
} from '@mui/material/MenuItem'

// ---| core |---
// ---| pages |---
// ---| screens |---
// ---| components |---
import Link, { LinkProps } from 'components/Link'

// ---| common |---
import { cn } from 'common/tools'

// ---| self |---
import css from './MenuItem.module.scss'

export type MenuItemProps = MuiMenuItemProps &
  Pick<LinkProps, 'start' | 'content' | 'end' | 'size' | 'iconSize' | 'align'>

/**
 * Component description.
 *
 * How to use
 * @example
 * <Item />
 */
export function MenuItem(props: MenuItemProps): JSX.Element {
  const {
    align,
    start,
    end,
    size,
    iconSize,
    content,
    className,
    ...otherProps
  } = props
  const _className = cn(css.MenuItem, className)

  return (
    <MuiMenuItem className={_className} {...otherProps}>
      <Link
        width='inherit'
        start={start}
        content={content}
        end={end}
        size={size}
        iconSize={iconSize}
        align={align}
      />
    </MuiMenuItem>
  )
}

MenuItem.displayName = 'MenuItem'

export default MenuItem
