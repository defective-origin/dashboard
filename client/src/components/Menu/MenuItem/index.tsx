export { default } from './MenuItem.component'
export * from './MenuItem.component'
