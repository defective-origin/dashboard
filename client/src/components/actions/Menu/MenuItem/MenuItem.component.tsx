import React from 'react'
import MuiMenuItem from '@mui/material/MenuItem'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
// ---| screens |---
// ---| components |---
import Button from 'components/actions/Button'
import Link from 'components/actions/Link'
import Divider from 'components/layouts/Divider'
import NavLink from 'components/actions/NavLink'
import { RepeatItem } from 'components/layouts/Repeat'
import { PopupTriggerOptions } from 'components/popups/Popup'

// ---| self |---
import css from './MenuItem.module.scss'

export const MENU_ACTION_MAP = {
  divider: Divider,
  button: Button,
  link: Link,
  nav: NavLink,
}

export type MenuItemProps = RepeatItem<typeof MENU_ACTION_MAP> & { triggerOptions?: PopupTriggerOptions }

/**
 * Component description.
 *
 * How to use
 * @example
 * <Item />
 */
export const MenuItem = (props: MenuItemProps) => {
  const { ref, triggerOptions: o, active, variant = 'button', className, children, ...otherProps } = props
  const _className = cn(css.MenuItem, className)
  const Tag = MENU_ACTION_MAP[variant] as React.FC<RepeatItem<typeof MENU_ACTION_MAP>>

  return (
    <MuiMenuItem ref={ref} className={_className}>
      {!children && Tag && (
        <Tag
          size='xxs'
          active={active ?? o?.isOn}
          color='primary'
          align='left'
          stretch
          {...otherProps}
        />
      )}

      {children}
    </MuiMenuItem>
  )
}

MenuItem.displayName = 'MenuItem'

export default MenuItem
