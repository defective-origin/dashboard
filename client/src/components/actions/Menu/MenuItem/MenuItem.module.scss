@import '/src/theme';

:root {
  // --ui-framework-color: var(--account-framework-color);
}

.MenuItem {
  padding: 0;
}
