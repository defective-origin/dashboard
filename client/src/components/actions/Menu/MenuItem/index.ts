/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './MenuItem.component'

/* INJECT_EXPORT_PLACE */
export * from './MenuItem.component'
