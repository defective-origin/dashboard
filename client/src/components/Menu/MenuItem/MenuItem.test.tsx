// ---| tests |---
import { render } from 'tests'

// ---| self |---
import MenuItem from './MenuItem.component'

describe('[MenuItem] component', () => {
  it('should render component', () => {
    const container = render(<MenuItem />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
