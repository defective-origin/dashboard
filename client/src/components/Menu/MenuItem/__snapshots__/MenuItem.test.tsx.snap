// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[MenuItem] component > should render component 1`] = `
<DocumentFragment>
  <li
    class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters action md _MenuItem_8f1bf0 css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
    role="menuitem"
    tabindex="-1"
  >
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </li>
</DocumentFragment>
`;
